torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -17,214 +17,220 @@ void initJitBackendBindings(PyObject* module) {
   // this function must be called like
   //
   //  torch._C._jit_to_backend("example_backend", module, spec)
-  auto m = py::handle(module).cast<py::module>();
-  m.def(
-      "_jit_to_backend",
-      [=](const std::string& backend_name,
-          const Module& orig_module,
-          const py::dict& method_compile_spec) {
-        const c10::QualifiedName qual_backend_name({"__torch__",
-                                                    "torch",
-                                                    "classes",
-                                                    detail::kBackendsNamespace,
-                                                    backend_name});
-        // TODO: Validate method_compile_spec.
-
-        // Clone orig_module to make sure backend transformation is
-        // functional.
-        auto cloned_module = orig_module.clone();
-
-        // Represents of a Type of Dict[str, Any].
-        auto any_dict_ty = DictType::create(StringType::get(), AnyType::get());
-
-        // Generate LoweredModule.
-        Module loweredModule(
-            "torch.jit." + backend_name + "LoweredModule",
-            get_python_cu(),
-            /*shouldMangle=*/true);
-
-        // Generate attributes.
-        // This is the original cloned and preprocessed module.
-        loweredModule.register_attribute(
-            "__processed_module",
-            AnyType::get(),
-            cloned_module._ivalue(),
-            /*is_param=*/false);
-
-        // This is for the method_compile_spec passed in to to_<backend> or
-        // loaded from an exported model.
-        loweredModule.register_attribute(
-            "__method_compile_spec",
-            any_dict_ty,
-            toIValue(method_compile_spec, any_dict_ty).toGenericDict(),
-            /*is_param=*/false);
-
-        // This is a pointer to a backend instance that is used to access
-        // compile and execute functions.
-        auto cls = getCustomClass(qual_backend_name.qualifiedName());
-        TORCH_INTERNAL_ASSERT(cls);
-        c10::intrusive_ptr<torch::CustomClassHolder> backend;
-        loweredModule.register_attribute(
-            "__backend", cls, IValue::make_capsule(backend));
-
-        // This is the list of opaque backend handles returned by
-        // backend.compile.
-        loweredModule.register_attribute(
-            "__handles",
-            any_dict_ty,
-            c10::impl::GenericDict(
-                any_dict_ty->getKeyType(), any_dict_ty->getValueType()),
-            /*is_param=*/false);
-
-        // Methods.
-
-        // This is a helper function for creating a new instance of the
-        // backend class.
-        static const auto create_backend_ct = CodeTemplate(R"(
+  auto codegen_lambda = [=](const std::string& backend_name,
```

Comment:
Mind commenting on what the meaningful change here is? I think the indentation shift made GH freak out

Reply:
I wrapped all the codegen in a lambda so I could do:
```
return wrap_cpp_module(codegen_lambda(args)))
```

instead of 
```
return wrap_cpp_module([](...){
// lots
// of
// code
});
```